### PR TITLE
Added auto-shutdown capabilities (disabled for now)

### PR DESCRIPTION
**Pull request by cptera**

### CODE REVIEW

File: dev/domain-controller/new-agent-vm/server2016-graphics-agent.json [hunk 1]
```diff
@@ -240,7 +241,8 @@
                 "videoDriverUrl": "[variables('videoDriverUrl')]",
                 "gitLocation": "[parameters('_artifactsLocation')]",
                 "sumoCollectorID": "[concat('CAMPOC-',resourceGroup().name,'-Agent-',subscription().subscriptionId)]",
-                "domainGroupToJoin": "[parameters('domainGroupToJoin')]"
+                "domainGroupToJoin": "[parameters('domainGroupToJoin')]",
+                "enableAutoShutdown": "[variables('enableAutoShutdown')]"
```
Comment (bill-dall) [inline dev/domain-controller/new-agent-vm/server2016-graphics-agent.json]: Ahh - So this default to empty string is 'disabled.'

File: dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1 [hunk 2]
```diff
@@ -372,6 +375,74 @@ Configuration InstallPCoIPAgent
...
+
+            TestScript = { 
+				$serviceName = "CAMIdleShutdown"
+				
+				if ([string]::IsNullOrWhiteSpace($using:enableAutoShutdown)) {
+					return $true
+				}
+
+				try {
+					$svc = Get-WmiObject -Class Win32_Service -Filter "Name='$ServiceName'"
```
Comment (bill-dall) [inline dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1]: Do we need an -erroraction stop inside the try block here? Traditionally I have seen that as needed for exception handling to work when DSC is invoked (which is different from standard script execution subtly).

File: dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1 [hunk 3]
```diff
@@ -372,6 +375,74 @@ Configuration InstallPCoIPAgent
...
+					throw $msg
+				}
+
+				$svc = Get-Service -Name $serviceName   
+
+				if ($svc.StartType -ne "Automatic") {
+					$msg = "try setting {0} Service start type to automatic." -f $serviceName
+					Write-Verbose $msg
+
+					Set-Service -name  $serviceName -StartupType Automatic
```
Comment (bill-dall) [inline dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1]: Similar comment to above -if 'Set-Service' has an error then even if you have error handling code after, DSC seems to sense the error (a disturbance in the force) and bork. This is a real PITA to test, naturally - did you see this handle these error conditions properly when you did your testing?

File: dev/domain-controller/new-agent-vm/server2016-graphics-agent.json (hunk 1 above)
Comment (fwang-teradici) [inline dev/domain-controller/new-agent-vm/server2016-graphics-agent.json]: It seems to use bool type and set default value to false is better choise

File: dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1 (hunk 2 above)
Comment (cptera) [inline dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1]: You're right - need to add the error action for the case where the service isn't installed

File: dev/domain-controller/new-agent-vm/server2016-graphics-agent.json (hunk 1 above)
Comment (cptera) [inline dev/domain-controller/new-agent-vm/server2016-graphics-agent.json]: done

File: dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1 (hunk 3 above)
Comment (cptera) [inline dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1]: I copied most of the Service setup stuff from another Script from the DSC. I haven't seen any errors related to this and we already try to get the service and check that it exists (by checking the StatType) so I don't think we need any additional error handling

**Pull request merged**